### DISCUSSION TITLE
Writing Platform-specific C# code using Partial class definitions not working with Uno Platform 4.8

### DISCUSSION THREAD

**Discussion by williamrousseau** [unoplatform/uno#12092] [Q&A]
I am trying to write Platform-specific C# code using Partial class definitions in a simple project using Uno Platform 4.8 and .NET 7.0 following the example showed in [Platform-specific C# code in Uno](https://platform.uno/docs/articles/platform-specific-csharp.html#partial-class-definitions) documentation, but there is an issue when adding the partial class tothe specific platform.

When calling the function created in the platform specific partial class from the partial class contained in shared, a building error occurs because the function does not exists.

How can we reference the Platform specific project to the shared project to access the other partial class part from the platform specific project?

## Here's my example:

Partial class in `SampleProject/MyPartialClass.cs`
```C#
namespace SampleProject
{
    public partial class MyPartialClass
    {
        public MyPartialClass() 
        {
            DoSomething();
        }
    }
}
```

Partial class in `SampleProject.Windows/MyPartialClass.cs`
```C#
namespace SampleProject
{
    public partial class MyPartialClass
    {
        public void DoSomething()
        {
            //Do something specific to Windows
        }
    }
}
```

Partial class in `SampleProject.WASM/MyPartialClass.cs`
```C#
namespace SampleProject
{
    public partial class MyPartialClass
    {
        public void DoSomething()
        {
            //Do something specific to WASM
        }
    }
}
```

Here's how the project is structured:

![image](https://user-images.githubusercontent.com/43794005/234696554-32e73ad3-5cc0-44d5-b7e4-fde4efe115b9.png)

Here's the error from Partial class in `SampleProject/MyPartialClass.cs`:

![image](https://user-images.githubusercontent.com/43794005/234697263-dbb7c563-4ada-4dca-89b3-6eaf82843ceb.png)
![image](https://user-images.githubusercontent.com/43794005/234697016-c5718422-df17-431a-95a2-f7cb06c15475.png)

My example is exactly the same as the example [here](https://platform.uno/docs/articles/platform-specific-csharp.html#partial-class-definitions), but it does not compile.

How to make it compile? Is it a problem known with Uno Platform 4.8 and .NET 7.0?

**Comment by jeromelaban** [comment]
The C# Partial classes feature only works in the same project. The documentation about it is dealing with shared projects, a structure that is not used anymore by the new project templates.

Are you trying to create a Wasm-specific class? If so, you may need to use inheritance. This will change in .NET 8 when the new `net8.0-browser` TargetFramework will be available.

**Reply by williamrousseau** [reply]
I am trying to to make a Windows-specific class, which need a specific Nuget package that doesn't compile with Android (System.Data.SQLite NuGet package). In order to use this package, I want to put the class under the MyProject.Windows folder.

How would it work with inheritance? How can I use the class under the platform-specific folder or only use the class I want with compiling in Windows?

**Reply by jeromelaban** [reply]
If you want to make a Windows specific class, you can use the `HAS_UNO` flag, which is not defined when building for WinAppSDK. Code conditional to `!HAS_UNO` will only run on WinAppSDK.
